packages/expect/src/jest-expect.ts

```diff
@@ -599,9 +599,9 @@ export const JestChaiExpect: ChaiPlugin = (chai, utils) => {
       const matcher = expected as any as AsymmetricMatcher<any>
       return this.assert(
         thrown && matcher.asymmetricMatch(thrown),
-        'expected error to match asymmetric matcher',
-        'expected error not to match asymmetric matcher',
-        matcher.toString(),
+        'expected thrown value #{act} to match asymmetric matcher: #{exp}',
+        'expected thrown value #{act} not to match asymmetric matcher: #{exp}',
```

Comment:
I went with using `#{act}` (short for "actual") here since currently `showDiff` is `false` (5th argument of `this.assert`) and the detail is not visible to users on terminal.

Looking at the history of whether enable `showDiff` or not, it thought this might be a performance concern of object comparison https://github.com/vitest-dev/vitest/pull/2828 https://github.com/vitest-dev/vitest/pull/3582 ("concordance" might expected to fixed this, but later reverted and `showDiff` became false), but considering `toEqual` and normal assertions showing diff already, would it be okay to put this `showDiff` back to `true` (i.e. default)?